rpc/system-rpc/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+use async_trait::async_trait;
+use jsonrpsee::{
+	core::RpcResult,
+	proc_macros::rpc,
+	tokio::sync::{
+		mpsc::{Receiver, Sender},
+		oneshot,
+	},
+	types::ErrorObject,
+};
+use serde::{Deserialize, Serialize};
+use sp_core::H256;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransactionState {
+	pub block_hash: H256,
+	pub block_height: u32,
+	pub tx_hash: H256,
+	pub tx_index: u32,
+	pub tx_success: bool,
+	pub pallet_index: u8,
+	pub call_index: u8,
+	pub is_finalized: bool,
+}
+
+#[derive(Clone)]
+pub struct Deps {
+	pub sender: TxStateSender,
+}
+
+pub type OneShotTxStateSender = oneshot::Sender<Vec<TransactionState>>;
+pub type TxStateReceiver = Receiver<(H256, bool, OneShotTxStateSender)>;
+pub type TxStateSender = Sender<(H256, bool, OneShotTxStateSender)>;
+pub type TxStateChannel = (H256, bool, OneShotTxStateSender);
+
+#[rpc(client, server)]
+pub trait TransactionState {
+	#[method(name = "system_transaction_state")]
```

Comment:
method name part should be camelCase as we're following this standard across all RPC's, also can we change the grouping of this RPC from system to something else to differentiate between default ones & our custom RPC's?

Reply:
Done. Thank you